modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/message/MessageServiceImpl.java

```diff
@@ -146,8 +154,18 @@ private void onMessageInternal(String consistentId, NetworkMessage msg) {
     /** {@inheritDoc} */
     @Override
     public void stop() {
-        if (lsnrs != null) {
-            lsnrs.clear();
+        if (!busyLock.enterBusy()) {
+            return;
```

Comment:
`SqlQueryProcessor` blocks the busy lock before calling this `stop()`, so it seem that we'll never be able to `enterBusy()` and we'll never set `stop` to `true`